scripts/battlefields/Cloister_of_Frost/waking_the_beast.lua

```diff
@@ -0,0 +1,79 @@
+-----------------------------------
+-- Area: Cloister of Frost
+-- BCNM: Waking the Beast
+-----------------------------------
+local cloisterOfFrostID = zones[xi.zone.CLOISTER_OF_FROST]
+-----------------------------------
+
+local content = BattlefieldQuest:new({
+    zoneId           = xi.zone.CLOISTER_OF_FROST,
+    battlefieldId    = xi.battlefield.id.WAKING_THE_BEAST_CLOISTER_OF_FROST,
+    canLoseExp       = false,
+    maxPlayers       = 18,
+    timeLimit        = utils.minutes(30),
+    index            = 3,
+    entryNpc         = 'IP_Entrance',
+    exitNpc          = 'Ice_Protocrystal',
+    requiredKeyItems = { xi.ki.RAINBOW_RESONATOR },
+
+    questArea = xi.questLog.OTHER_AREAS,
+    quest     = xi.quest.id.otherAreas.WAKING_THE_BEAST,
+})
+
+function content:onEventFinishWin(player, csid, option, npc)
+    npcUtil.giveKeyItem(player, xi.ki.EYE_OF_FROST)
+end
+
+content.groups =
+{
+    {
+        -- avatar
+        mobIds =
+        {
+            { cloisterOfFrostID.mob.SHIVA_PRIME_WTB      },
+            { cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 5  },
+            { cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 10 },
+        },
+
+        allDeath = function(battlefield, mob)
+            -- when avatar defeated then all elementals should also die
+            for i = 1, 4 do
+                local elemental = GetMobByID(mob:getID() + i)
+                if elemental and elemental:isAlive() then
+                    elemental:setHP(0)
+                end
+            end
+
+            battlefield:setStatus(xi.battlefield.status.WON)
+        end,
+    },
+
+    {
+        -- elementals
+        mobIds =
+        {
+            {
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 1,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 2,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 3,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 4,
+            },
+
+            {
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 6,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 7,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 8,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 9,
+            },
+
+            {
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 11,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 12,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 13,
+                cloisterOfFrostID.mob.SHIVA_PRIME_WTB + 14,
+            },
+        },
+    },
+}
```

Comment:
You should be able to use names rather than ids here. It should automatically figure out the stride for the mobs and which mobs belong to each battlefield area. I'd try it with one and test it. It should work. If it does work then you can simplify them all!
```lua
{
    mobs = { "Shiva_Prime_WTB" }
    allDeath = ...
},
{
    mobs = { "Ice_Elemental" }
}
```

Reply:
> You should be able to use names rather than ids here. It should automatically figure out the stride for the mobs and which mobs belong to each battlefield area. I'd try it with one and test it. It should work. If it does work then you can simplify them all!
> 
> ```lua
> {
>     mobs = { "Shiva_Prime_WTB" }
>     allDeath = ...
> },
> {
>     mobs = { "Ice_Elemental" }
> }
> ```

Thanks, yep, this should now be fixed.